### PR TITLE
Query and Update "animals" Table PR

### PR DESCRIPTION
**Pull request by Aseadd**
### Commits made for this PR

- [ ] Alter Table and add new column species
- [ ] Insert new data to the table 
- [ ] Update column transaction
- [ ] Delete data with transaction
- [ ] Make queries by applying the aggregate concept
Screenshot from 2022-10-19 01-25-47
Screenshot from 2022-10-19 01-22-58
Screenshot from 2022-10-19 01-18-50
Screenshot from 2022-10-19 01-08-24
Screenshot from 2022-10-19 00-48-12
Screenshot from 2022-10-19 00-47-09
Screenshot from 2022-10-19 00-44-49
Screenshot from 2022-10-18 23-56-37
Screenshot from 2022-10-18 23-55-02
Screenshot from 2022-10-18 23-53-29
Screenshot from 2022-10-18 23-47-17
Screenshot from 2022-10-18 23-42-44
Screenshot from 2022-10-18 23-42-03
Screenshot from 2022-10-18 23-25-41
Screenshot from 2022-10-18 23-23-56
Screenshot from 2022-10-18 23-19-17
Screenshot from 2022-10-18 23-18-21

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -4,4 +4,7 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Gabumon', '2018/11/15', 2, true, 8);
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Pikachu', '2021/01/07', 1, false, 15.04);
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Devimon', '2017/05/12', 5, true, 11);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, wight_kg) VALUES ('Charmander', '2020/02/08', 0, false, -11), 
+('Plantmon','2021/03/15', 2, true, -5.7), ('Squirtle', '1993/04/02', 3, false, -12.13), ('Angemon', '2005/06/12', 1, true, -45),
```
Comment (sja-thedude) [inline data.sql:8]: - [ ] Kindly make sure that the date of birth for Plantmon is correct. It should be 2021/11/15.

File: data.sql [hunk 2]
```diff
@@ -4,4 +4,7 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Gabumon', '2018/11/15', 2, true, 8);
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Pikachu', '2021/01/07', 1, false, 15.04);
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Devimon', '2017/05/12', 5, true, 11);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, wight_kg) VALUES ('Charmander', '2020/02/08', 0, false, -11), 
+('Plantmon','2021/03/15', 2, true, -5.7), ('Squirtle', '1993/04/02', 3, false, -12.13), ('Angemon', '2005/06/12', 1, true, -45), 
+('Boarmon','2005/07/06', 7, true, 20.4), ('Blossom', '1998/10/13', 3, true, 17), ('Ditto', '2022/05/14', 4, true, 22);
```
Comment (sja-thedude) [inline data.sql]: - [ ] Kindly make sure that the date of birth for Boarmon is correct. It should be 2005/06/07.

File: queries.sql [hunk 3]
```diff
@@ -8,3 +8,42 @@ SELECT * FROM animals WHERE neutered = true;
...
+UPDATE animals SET species = 'pokemon' WHERE name NOT  LIKE '%mon';
+SELECT * FROM animals;
+COMMIT;
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK; 
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022/01/01';
+DELETE 1
```
Comment (sja-thedude) [inline queries.sql]: - [ ] You have added an extra line 36, kindly delete it.

File: queries.sql [hunk 4]
```diff
@@ -8,3 +8,42 @@ SELECT * FROM animals WHERE neutered = true;
...
+SELECT * FROM animals;
+SAVEPOINT my_savepoint1;
+UPDATE animals SET wight_kg = wight_kg*-1;
+ROLLBACK TO SAVEPOINT my_savepoint1;
+UPDATE animals SET wight_kg = wight_kg*-1 WHERE wight_kg < 0;
```
Comment (sja-thedude) [inline queries.sql:40]: - [ ] Kindly add commit after the transaction.

File: queries.sql [hunk 5]
```diff
@@ -8,3 +8,42 @@ SELECT * FROM animals WHERE neutered = true;
...
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(escape_attempts) FROM animals WHERE escape_attempts =0;
+SELECT AVG(wight_kg) FROM animals;
+SELECT name  FROM animals WHERE escape_attempts = ( SELECT MAX (escape_attempts) FROM animals WHERE neutered = true OR neutered = false );
+SELECT species, MAX(wight_kg) FROM animals GROUP BY species; 
+SELECT species, MAX(wight_kg), MIN(wight_kg) FROM animals GROUP BY species; 
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990/01/01' AND '2000/01/01' GROUP BY species;
```
Comment (sja-thedude) [inline queries.sql]: - [ ] Kindly make sure the end date is 31 Dec 2000 as the query is to find an average number of escape attempts per animal type of those born between 1990 and 2000, so the entire 2000 year is included.

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @Aseadd, This is @sja-thedude your code reviewer.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

768070f3-bc9a-4aba-b8bc-25793e00ea1e_text

## Highlights
✔️ Professional Readme
✔️ Correct gitflow

## Required 
- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 3 above)
Comment (Aseadd) [reply queries.sql]: Corrected with big thanks!!

File: queries.sql (hunk 5 above)
Comment (Aseadd) [reply queries.sql]: Changed to '2000/12/31'

Comment (sja-thedude) [reply queries.sql]: @Aseadd Good job, However, you have not corrected a small part. https://github.com/Aseadd/vet-clinic-db/pull/3#discussion_r998777436 . If you could do that I will approve.

File: queries.sql (hunk 4 above)
Comment (Aseadd) [reply queries.sql:40]: COMMIT Transaction added on line  #41

File: queries.sql (hunk 5 above)
Comment (Aseadd) [reply queries.sql]: COMMIT Transaction added  on line #41

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @Aseadd, This is @sja-thedude your code reviewer.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

200w

 ## Highlights
✔️ Good job

 ## Required Changes ♻️
- [ ] Kindly make sure to implement one small issue https://github.com/Aseadd/vet-clinic-db/pull/3#discussion_r998777436

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 4 above)
Comment (Aseadd) [reply queries.sql:40]: Screenshot from 2022-10-19 02-40-08

**Comment by Aseadd** [comment]
Screenshot from 2022-10-19 02-40-08
Made the change even before this review Please check it well!!

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @Aseadd, 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well-implemented requirements.

### Optional suggestions :arrow_down_small: 

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

* N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by zieeco — COMMENTED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @Aseadd , 


68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966


Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡ 

✅ Descriptive PR 👍 
✅ Linters are passing 👏 


## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**